new_platforms/vsextension/ProjectWizard/WizardImplementation.cs

```diff
@@ -72,22 +117,62 @@ private bool SetOETADevKitPath(Dictionary<string, string> replacementsDictionary
 
                     // Get the path of specified file.
                     string filePath = openFileDialog.FileName;
-                    folder = Path.GetFullPath(Path.Combine(filePath, "..\\.."));
+                    oeFolder = Path.GetFullPath(Path.Combine(filePath, ".."));
                 }
             }
+            replacementsDictionary.Add("$OELibPath$", GetUnixPath(oeFolder));
 
-            // We now have the full path in Windows format, but we need to convert it to Unix format.
-            string root = Path.GetPathRoot(folder).ToLower();
-            string relativeFolder = folder.Substring(root.Length).Replace('\\', '/');
-            string drive = root.Substring(0, 1);
-            string unixPath = "/mnt/" + drive + "/" + relativeFolder;
+            // Now get the path to the associated TA Dev Kit, which should be under
+            // devkit or export-ta_arm64 or export-ta_arm32.
+            string taDevKitFolder = Path.Combine(oeFolder, "devkit");
+            if (!Directory.Exists(taDevKitFolder))
+            {
+                taDevKitFolder = Path.Combine(oeFolder, "export-ta_arm64");
+            }
+            if (!Directory.Exists(taDevKitFolder))
+            {
+                taDevKitFolder = Path.Combine(oeFolder, "export-ta_arm32");
+            }
+            replacementsDictionary.Add("$OETADevKitPath$", GetUnixPath(taDevKitFolder));
 
-            replacementsDictionary.Add("$OETADevKitPath$", unixPath);
+            bool hardwareFloatSupported;
+            bool is32Bit;
+            GetConfFlags(taDevKitFolder, out hardwareFloatSupported, out is32Bit);
 
-            string buildFlavor = board.Split('\\')[0];
-            replacementsDictionary.Add("$OpteeBuildFlavor$", buildFlavor);
+            string opteeCompilerFlavor;
+            if (!is32Bit)
+            {
+                opteeCompilerFlavor = "aarch64-linux-gnu-";
+            }
+            else
+            {
+                if (hardwareFloatSupported)
+                {
+                    opteeCompilerFlavor = "arm-linux-gnueabihf-";
+                }
+                else
+                {
+                    opteeCompilerFlavor = "arm-linux-gnueabi-";
 
-            string opteeCompilerFlavor = (buildFlavor == "vexpress-qemu_virt") ? "arm-linux-gnueabihf-" : "aarch64-linux-gnu-";
+                    // Output a warning so the developer knows they won't get hardware float support.
```

Comment:
Consider moving the warning only if we fail to detect hardware float support, on line 66. If the board being targeted does not have hardware float support, the user will get a false warning.

Reply:
There's no way to check whether the board has hardware float support, only whether the user has configured it in conf.mk, right?   That's why I didn't consider the warning to be "false".  I can move it if you still feel strongly.

Reply:
Updated per our discussion